terraform/variables.go

```diff
@@ -1,25 +1,64 @@
 package terraform
 
 import (
+	"strings"
+
 	"github.com/hashicorp/hcl/v2"
 	"github.com/zclconf/go-cty/cty"
 )
 
-func extractFromFile(file *hcl.File) (VariableMap, hcl.Diagnostics) {
-	// TODO: remove temporary dummy variables
-	// implement "var" when we do input variables and default values
+type VariableMap map[string]cty.Value
+
+func extractFromFile(filename string, file *hcl.File) (VariableMap, hcl.Diagnostics) {
+	// TODO: remove temporary dummy locals
 	// implement "local" when we do local values
+	varMap := VariableMap{}
+	var hclDiags hcl.Diagnostics
+
+	if strings.HasSuffix(filename, TF) {
+		varMap, hclDiags = extractFromTfFile(filename, file)
+		if hclDiags.HasErrors() {
+			return VariableMap{}, hclDiags
+		}
+	}
+
 	return VariableMap{
-		"var": cty.ObjectVal(VariableMap{
-			"dummy": cty.StringVal("dummy_value"),
-		}),
+		"var": cty.ObjectVal(varMap),
 		"local": cty.ObjectVal(VariableMap{
-			"dummy": cty.StringVal("dummy_local"),
+			"dummy": cty.StringVal("dummy"),
 		}),
-	}, nil
+	}, hclDiags
 }
 
-type VariableMap map[string]cty.Value
+func extractFromTfFile(filename string, file *hcl.File) (VariableMap, hcl.Diagnostics) {
+	varMap := VariableMap{}
+	bodyContent, hclDiags := file.Body.Content(tfFileSchema)
+
+	handleDiagnostics("Validation issue", hclDiags, filename)
+	if hclDiags.HasErrors() {
+		return VariableMap{}, hclDiags
+	}
+
+	for _, block := range bodyContent.Blocks {
```

Comment:
Is this going to run on all types of blocks? A file might have more than just `variable` blocks, so for example if I add the following test:
```
{
			name: "Non-variable block",
			input: TestInput{
				fileName: "test.tf",
				fileContent: `
provider "google" {
  project = "acme-app"
  default  = "us-central1"
}`,
			},
			expected: VariableMap{
				"var": cty.ObjectVal(VariableMap{
					"google": cty.StringVal("us-central1"),
				}),
				"local": cty.ObjectVal(VariableMap{
					"dummy": cty.StringVal("dummy"),
				}),
			},
		},
```

It passes although `google` is not a variable